api/src/infrastructure/postgres-survey-data.repository.ts

```diff
@@ -0,0 +1,107 @@
+import { DataSource, EntityManager } from 'typeorm';
+import { ISurveyDataRepository } from '@api/infrastructure/survey-data-repository.interface';
+import { Inject, Logger } from '@nestjs/common';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { WidgetDataFilters } from '@shared/dto/widgets/widget-data-filter';
+import { SectionWithDataWidget } from '@shared/dto/sections/section.entity';
+import { SQLAdapter } from '@api/infrastructure/sql-adapter';
+import { WidgetQuestionMap } from '@shared/dto/widgets/widget-question-map';
+import {
+  BaseWidgetWithData,
+  WidgetData,
+} from '@shared/dto/widgets/base-widget-data.interface';
+
+export class PostgresSurveyDataRepository implements ISurveyDataRepository {
```

Comment:
I see the point of this adapter, but we need to have our feet in the ground. It could be useful, maybe to have an adapter for an ORM, but assuming that we need to be able to switch database engine it's not realistic